### ISSUE TITLE
cannot read property 'autoAriaDisabled' of undefined

### ISSUE THREAD

**Issue by adamweaver** [isaacHagoel/svelte-dnd-action#191]
Hi Isaac,

My guess is that some globals aren't cleared when dndzone is destroyed; presumably since the library can't easily know that there aren't other dndzones mounted elsewhere. Or my guess could be wrong!

Adjusting the simple REPL example to add a show/hide toggle:

```
<style>
	section {
		width: 12em;
		padding: 1em;
		height: 7.5em;
	}
	div {
		height: 1.5em;
		width: 10em;
		text-align: center;
		border: 1px solid black;
		margin: 0.2em;
		padding: 0.3em;
	}
</style>
<script>
	import {dndzone} from 'svelte-dnd-action'; 
	function handleSort(e) {
		items = e.detail.items;
	}
	let show = true;
	let items = [
		{id:1, title: 'I'},
		{id:2, title: 'Am'},
		{id:3, title: 'Yoda'}
	];
</script>
<button on:click={() => show = !show}>toggle</button>
{#if show}
<section use:dndzone={{items}} on:consider={handleSort} on:finalize={handleSort}>
	{#each items as item(item.id)}
		<div>
			{item.title}	
		</div>
	{/each}
</section>
{/if}
```

If you activate keyboard nav (e.g. with the spacebar), when the `<section>` is hidden (via the "toggle" button) and reshown, we have some errors:

```
VM27:1770 Uncaught TypeError: Cannot read property 'autoAriaDisabled' of undefined
    at handleDrop$1 (eval at handle_message (about:srcdoc:13), <anonymous>:1770:41)
    at globalClickHandler (eval at handle_message (about:srcdoc:13), <anonymous>:1730:13)
```

Is this an easy fix, or is there an API func to let the library know that all state should be cleared?

**Comment by isaacHagoel** [comment]
@adamweaver I see why it happens. the dnd zone gets unregistered (removed from the global data structure) but the drop handler isn't removed because i didn't expect the rug to be pulled from underneath this way :) 
What is the correct behaviour when the zone the item is in is removed mid drag?
I could just leave it there I guess (meaning it disappears with the zone). Is this acceptable?

**Comment by adamweaver** [comment]
Hi Isaac, for my use case, the drop handler should be removed. But I presume you don't want to _always_ do that in case other use cases expect the item to still be available... I guess? Though to be honest I can't think why you'd want a mid-drag element to stick around when its old parent is binned.

**Comment by isaacHagoel** [comment]
I'll think about it some more and aim at having it fixed before the end of the week.

**Comment by adamweaver** [comment]
<img width="1280" alt="Screen Shot 2020-12-15 at 12 44 04 pm" src="https://user-images.githubusercontent.com/1251895/102172190-42858400-3ed3-11eb-9cce-6357a947816d.png">

Just to show why I was doing something wacky like pulling the rug away .... toggling the "drag-and-drop layout design" mode to "edit" mode destroys all dndzones to swap in visual editors. I can work-around by hiding the drag-n-drop rather than destroying them, of course.

**Comment by isaacHagoel** [comment]
hi @adamweaver fixed in version 0.6.24. pls let me know if there is anything else i can assist with.
